programs/belauncher/main.c

```diff
@@ -56,7 +57,37 @@ int WINAPI wWinMain(HINSTANCE hInst, HINSTANCE hPrevInst, LPWSTR cmdline, int cm
     }
     while ((config = strchr(config, '\n')) && *(config++));
 
-    if (arch_64_exe && (sizeof(void *) == 8 || (IsWow64Process(GetCurrentProcess(), &wow64) && wow64)))
+    game_cmd = cmdline;
+
+    /* Check if -exe argument was passed, which will override .ini entry */
+    cmd_exe_arg = wcsstr(cmdline, L"-exe ");
+    if (cmd_exe_arg)
+    {
+        if (wcslen(cmd_exe_arg) < 6)
+            goto no_exe_arg;
+        exe_arg_uses_quotation = cmd_exe_arg[5] == '\"';
+        cmd_exe_arg_end = wcschr(&cmd_exe_arg[6], exe_arg_uses_quotation ? L'\"' : L' ');
+        if (!cmd_exe_arg_end)
+            goto no_exe_arg;
+
+        /* Extract the game executable */
+        exe_arg_len = cmd_exe_arg_end - cmd_exe_arg - 6;
+        arg_exe = HeapAlloc(GetProcessHeap(), 0, exe_arg_len + 1);
+        WideCharToMultiByte(CP_ACP, 0, cmd_exe_arg + 6, exe_arg_len, arg_exe, exe_arg_len, NULL, NULL);
```

Comment:
This will skip the first character of the exe if there is no quotation mark.  In my opinion, we should just increment cmd_exe_arg to the start of the exe name when we are looking for it (incrementing it one more when we see a quotation), and use that.  We can the start again for rebuilding the command line.

Reply:
Done.